pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -551,16 +623,16 @@ func (nc *NodeController) MonitorNodeStatus() error {
 			continue
 		}
 
-		readyCondition := nc.getCondition(node, api.NodeReady)
+		readyCondition := nc.getCondition(&node.Status, api.NodeReady)
```

Comment:
I'm not sure I understood correctly, but if I do I guess I just need to return readyCondition from tryXXX to keep "old" behaviour.